core/src/main/scala/org/bitcoins/core/api/dlcoracle/OracleEvent.scala

```diff
@@ -168,7 +169,35 @@ case class CompletedDigitDecompositionV0OracleEvent(
     outcomes: Vector[DLCAttestationType],
     attestations: Vector[FieldElement])
     extends CompletedOracleEvent
-    with DigitDecompositionV0OracleEvent
+    with DigitDecompositionV0OracleEvent {
+
+  val outcomeBase10: Long = {
+    val (digits, positive) = {
+      eventDescriptorTLV match {
+        case _: SignedDigitDecompositionEventDescriptor =>
+          val positive = outcomes.head
```

Comment:
It seems like we need to make some dedicated types rather than using `outcomes: Vector[DLCAttestationType]` everywhere. 

That way we can at least get some type safety to know the `outcomes: Vector[DLCAttestationType]` doesn't correspond to a enum event for instance. 

it also allows us to get around this nasty casting if I understand the problem. 

If I understand correctly, we could even have 

`outcomes = Vector(RangeAttestation, EnumAttestation, DigitDecompAttestation)` 

basically the vector not being internally consistent with the type since all of these things are subtypes of `DLCAttestationType`.